### PR TITLE
[1LP][RFR] Remove old setup_a_provider

### PR DESCRIPTION
**Pull request by jkrocil**
WIP - a follow-up commit will enforce (hopefully) all provider setup through fixtures. That's the plan, anyway.

Edit: will be a separate PR. This one is ready.

### CODE REVIEW

File: cfme/rest/gen_data.py [hunk 1]
```diff
@@ -238,7 +240,7 @@ def _finished():
 
 
 def a_provider():
```
Comment (psav) [inline cfme/rest/gen_data.py]: Wondering if we should normalize all these
we have ```a_provider``` here and ```setup_a_provider``` else where. I guess we should look at what the fixture actually does. ```a_provider``` is possibly better than ```setup_a_provider```, thoughts?

Comment (jkrocil) [inline cfme/rest/gen_data.py]: Yes. There is also stuff like `setup_first_provider`, `setup_single_provider`, `vmware_provider` and also just `provider`. All of that will (like I said "hopefully) come together nicely and be unified.

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Comment by dajoRH** [comment]
Would you mind rebasing this Pull Request against latest master, please? :trollface:
*CFME QE Bot*

**Pull request merged**